naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2474,6 +2480,29 @@ impl Parser {
 
         let mut lexer = Lexer::new(source);
         let mut tu = ast::TranslationUnit::default();
+
+        // Parse directives.
+        #[allow(clippy::never_loop)]
+        loop {
+            if let Ok(ident) = lexer.peek_ident() {
+                self.push_rule_span(Rule::Directive, &mut lexer);
+                if let Some((kind, name)) = DirectiveKind::from_ident(ident.name) {
+                    match kind {
+                        DirectiveKind::Unimplemented(unimplemented) => {
+                            return Err(Error::DirectiveNotYetImplemented {
+                                kind: name,
+                                span: ident.span,
+                                tracking_issue_num: unimplemented.tracking_issue_num(),
+                            })
```

Comment:
nitpick: I think I should consolidate the `kind` (`&str`) and `tracking_issue_num` fields into an `kind` of the unimplemented `enum`. The `&str` and `u16` can be derived from the `enum`.